src/NumericalAlgorithms/Convergence/HasConverged.hpp

```diff
@@ -0,0 +1,95 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <boost/none.hpp>
+#include <boost/optional.hpp>
+#include <cstddef>
+#include <iosfwd>
+
+#include "NumericalAlgorithms/Convergence/Criteria.hpp"
+#include "NumericalAlgorithms/Convergence/Reason.hpp"
+#include "Options/Options.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+namespace Convergence {
+
+/*!
+ * \brief Determine whether the \p criteria are met.
```

Comment:
It's not clear to me this function should be public vs having people use the `HasConverged` class. @nilsleiffischer do you have any reasons to prefer it being public?

Reply:
discussed this offline and decided to keep as-is